indexing-hadoop/src/main/java/io/druid/indexer/IndexGeneratorJob.java

```diff
@@ -382,8 +383,18 @@ protected void reduce(
           context.progress();
           final InputRow inputRow = index.formatRow(parser.parse(value.toString()));
           allDimensionNames.addAll(inputRow.getDimensions());
-
-          int numRows = index.add(inputRow);
+          int numRows;
+          try {
+            numRows = index.add(inputRow);
+          } catch (ParseException e) {
+            if (config.isIgnoreInvalidRows()) {
+              log.info("Ignoring invalid row [%s] due to parsing error: %s", value.toString(), e.getMessage());
+              context.getCounter(HadoopDruidIndexerConfig.IndexJobCounters.INVALID_ROW_COUNTER).increment(1);
```

Comment:
IIRC there is a way to print out the file the error occurred in. That might be a helpful part of the error message?

(general thought, not required)